os/drivers/pm/pm.c

```diff
@@ -117,6 +117,9 @@ static int pm_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 			ret = pm_timedsuspend(((pm_suspend_arg_t *)arg)->domain, ((pm_suspend_arg_t *)arg)->timer_interval);
 		}
 		break;
+	case PMIOC_DOMAIN_REGISTER:
+		ret = pm_domain_register((char *) arg);
```

Comment:
the return of ioctl may be confused with the error code. Please use argument for return domain id

And please add desciption into api description

Reply:
Done